workers/unity/Packages/com.improbable.gdk.core/Worker/ViewDiff.cs

```diff
@@ -0,0 +1,307 @@
+            }
+        }
+
+        public void AddComponent<T>(T component, long entityId, uint componentId) where T : ISpatialComponentUpdate
+        {
+            if (!componentIdToComponentStorage.TryGetValue(componentId, out var storage))
+            {
+                throw new ArgumentException("Unknown component ID");
+            }
+
+            ((IDiffComponentAddedStorage<T>) storage).AddEntityComponent(entityId, component);
+        }
+
+        public void RemoveComponent(long entityId, uint componentId)
+        {
+            if (!componentIdToComponentStorage.TryGetValue(componentId, out var storage))
+            {
+                throw new ArgumentException("Unknown component ID");
+            }
+
+            storage.RemoveEntityComponent(entityId);
+        }
+
+        public void SetAuthority(long entityId, uint componentId, Authority authority)
+        {
+            if (!componentIdToComponentStorage.TryGetValue(componentId, out var authorityStorage))
+            {
+                throw new ArgumentException("Unknown component");
+            }
+
+            ((IDiffAuthorityStorage) authorityStorage).AddAuthorityChange(
+                new AuthorityChangeReceived(authority, new EntityId(entityId)));
+
+            // Remove received command requests if authority has been lost
+            if (authority == Authority.NotAuthoritative)
+            {
+                if (componentIdToCommandIdToStorage.TryGetValue(componentId, out var commandIdToStorage))
+                {
+                    foreach (var storage in commandIdToStorage)
+                    {
+                        storage.Value.RemoveEntityComponent(entityId);
+                    }
+                }
+            }
+        }
+
+        // public void MarkAuthorityTemporarilyLost(long entityId, uint componentId)
+        // {
+        //     if (!componentIdToComponentStorage.TryGetValue(componentId, out var authorityStorage))
+        //     {
+        //         throw new ArgumentException("Unknown component");
+        //     }
+        //
+        //     ((IDiffAuthorityStorage) authorityStorage).AddAuthorityChange(
+        //         new AuthorityChangeReceived(authority, new EntityId(entityId)));
+        //
+        //     authorityLostTemporarily.Add(new EntityComponent(entityId, componentId));
+        // }
+
+        public void AddComponentUpdate<T>(T update, long entityId, uint componentId, uint updateId)
+            where T : ISpatialComponentUpdate
+        {
+            if (!componentIdToComponentStorage.TryGetValue(componentId, out var storage))
+            {
+                throw new ArgumentException("Unknown component");
+            }
+
+            ((IDiffUpdateStorage<T>) storage).AddUpdate(new ComponentUpdateReceived<T>(update, new EntityId(entityId),
+                updateId));
+        }
+
+        public void AddEvent<T>(T ev, long entityId, uint componentId, uint updateId) where T : IEvent
```

Comment:
ev -> event

Reply:
event is a keyword. Can use `@event` or make it something else.

Reply:
A true, then its fine I think